sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/FunctionAppImpl.java

```diff
@@ -171,13 +174,49 @@ public FunctionAppImpl withLatestRuntimeVersion() {
         return withRuntimeVersion("latest");
     }
 
+    @Override
+    Mono<SiteInner> submitSite(SiteInner site) {
+        if (isFunctionAppOnACA()) {
+            return createOrUpdateInner(site);
+        } else {
+            return super.submitSite(site);
+        }
+    }
+
+    @Override
+    Mono<SiteInner> submitSite(SitePatchResourceInner siteUpdate) {
+        if (isFunctionAppOnACA()) {
+            return updateInner(siteUpdate);
+        } else {
+            return super.submitSite(siteUpdate);
+        }
+    }
+
     @Override
     Mono<Indexable> submitAppSettings() {
         if (storageAccountCreatable != null && this.taskResult(storageAccountCreatable.key()) != null) {
             storageAccountToSet = this.taskResult(storageAccountCreatable.key());
         }
         if (storageAccountToSet == null) {
             return super.submitAppSettings();
+        } else if (isFunctionAppOnACA()) {
+            return storageAccountToSet
+                .getKeysAsync()
+                .flatMap(storageAccountKeys -> {
+                    StorageAccountKey key = storageAccountKeys.get(0);
+                    String connectionString = ResourceManagerUtils
+                        .getStorageConnectionString(storageAccountToSet.name(), key.value(),
+                            manager().environment());
+                    addAppSettingIfNotModified(SETTING_WEB_JOBS_STORAGE, connectionString);
+                    addAppSettingIfNotModified(SETTING_WEB_JOBS_DASHBOARD, connectionString);
+                    return FunctionAppImpl.super.submitAppSettings();
+                }).then(
+                    Mono
+                        .fromCallable(
+                            () -> {
+                                resetStorageInfo();
+                                return this;
+                            }));
```

Comment:
Any difference from the code below? Can we extract them to a shared method?

Reply:
For Function App on ACA, appServicePlanId() is null, resulting in NPE for this.manager().appServicePlans().getByIdAsync(this.appServicePlanId()). 
And two app settings `WEBSITE_CONTENTAZUREFILECONNECTIONSTRING` and `WEBSITE_CONTENTSHARE` are not needed.

We should be able to combine them into one.

Reply:
Ok. If more difference, you can keep it, but better to add comment about the difference, so dev would know some configure is omitted in purpose.

Reply:
Combined the two parts into one.